File: LinearSearch/linear.cpp
#include <iostream>

#define cin std::cin
#define cout std::cout

int linearSearch(int arr[], int n, int find) {
    for(int i = 0; i < n; i++) {
        if(arr[i] == find) {
            return i;
        }
    }
    return -1;
}

int main() {
    int n;

    cout << "Enter number of elements: ";
    cin >> n;

    int arr[n], find, ans;

    for(int i = 0; i < n; i++) {
        cout << "Enter an element: ";
        cin >> arr[i];
    }

    cout << "Enter an element to find: ";
    cin >> find;

    ans = linearSearch(arr, n, find);

    if(ans == -1) {
        cout << "Element not found";
    }
    else {
        cout << "Element found at index: " << ans + 1;
    }
}

File: LinearSearch/linear.lua
function linearSearch(arr, find)
    for i = 1, #arr do
        if arr[i] == find then
            return i
        end
    end
    return -1
end

function main()
    io.write("Enter number of elements: ")
    local n = io.read("*n")

    local arr = {}

    for i = 1, n do
        io.write("Enter an element: ")
        arr[i] = io.read("*n")
    end

    io.write("Enter an element to find: ")
    local find = io.read("*n")

    local ans = linearSearch(arr, find)

    if ans == -1 then
        print("Element not found")
    else
        print("Element found at index: "..ans)
    end
end

main()
